source/_UIAHandler.py

```diff
@@ -59,13 +59,15 @@
 	"SysListView32",
 	"EXCEL7",
 	"Button",
-	# #7497: Windows 10 Fall Creators Update has an incomplete UIA implementation for console windows, therefore for now we should ignore it.
-	# It does not implement caret/selection, and probably has no new text events.
-	"ConsoleWindowClass",
 	# #8944: The Foxit UIA implementation is incomplete and should not be used for now.
 	"FoxitDocWnd",
 ]
 
+# #7497: Windows 10 Fall Creators Update has an incomplete UIA implementation for console windows, therefore for now we should ignore it.
+# It does not implement caret/selection, and probably has no new text events.
+if not config.conf['UIA']['consoleUIA']:

```

Comment:
It would be good if this could be checked at runtime so that changing the option in the Advanced settings panel allows UIA to be used in consoles without restarting NVDA. I would suggest adding an isBadUIAWindow function to _UIAHandler, which should check badUIAWindowClassNames and also use this code. Then in _isUIAWindowHelper, isBadUIAWindow should be called instead of checking badUIAWindowClassNames directly.